src/router/index.js

```diff
@@ -0,0 +1,70 @@
+import Vue from 'vue'
+import VueRouter from 'vue-router'
+import Home from '../views/Home.vue'
+import ArmyDetails from '../views/ArmyDetails.vue'
+import Battle from '../views/Battle.vue'
+import UploadArmy from '../views/UploadArmy.vue'
+import BattleHistory from '../views/BattleHistory.vue'
+import BattleResult from '../views/BattleResult.vue'
+Vue.use(VueRouter)
+
+const routes = [
```

Comment:
please consider route naming this if it makes sense.

/droidarmy .                 -> /droid
/troppersarmy             -> /troopers
/addtroppersarmy      -> /troopers/add
/addroidarmy .            -> /droid/add
/battleground .           -> /battle
/battlehistory              -> /battle/history
/battleresult                -> /battle/result